fluss-server/src/test/java/com/alibaba/fluss/server/coordinator/StopReplicaITCase.java

```diff
@@ -156,9 +156,14 @@ private List<Path> assertReplicaExistAndGetTableOrPartitionDirs(
             assertThat(replica.getLogTablet().getLogDir()).exists();
             if (isKvTable) {
                 // wait the replica become leader, so that we can get the kv tablet
-                replica = FLUSS_CLUSTER_EXTENSION.waitAndGetLeaderReplica(replica.getTableBucket());
-                assertThat(replica.getKvTablet()).isNotNull();
-                assertThat(replica.getKvTablet().getKvTabletDir()).exists();
+                Replica kvReplica =
+                        FLUSS_CLUSTER_EXTENSION.waitAndGetLeaderReplica(replica.getTableBucket());
+                retry(
+                        Duration.ofMinutes(1),
+                        () -> {
+                            assertThat(kvReplica.getKvTablet()).isNotNull();
+                            assertThat(kvReplica.getKvTablet().getKvTabletDir()).exists();
+                        });
```

Comment:
This change looks strange to me. If the `kvReplica` is already leader, why the `kvtablet` is not ready? It means the leader flag in `Replica` is incorrect. 

Should we move `leaderReplicaIdOpt.set(localTabletServerId);` after `onBecomeNewLeader()`?